### DISCUSSION TITLE
Support for Table of Contents (ToC)?

### DISCUSSION THREAD

**Discussion by borekb** [notaapp/talk#90] [Feature Requests]
If I'm not mistaken, Caret used to support `[[toc]]` as a way to generate table of contents (ToC) which was quite useful. It's not necessary to have a special syntax for it but I'd appreciate a command like **Generate table of contents**.

The _Markdown All in One_ extension for VSCode [does ToC nicely](https://github.com/yzhang-gh/vscode-markdown#table-of-contents).

**Comment by astoilkov** [comment]
We should make that possible through extensibility. Maybe [Custom Commands](https://nota.md/docs/custom-commands.html) should support this scenario.

Currently, you can rewrite the entire file and place a ToC inside but that may be too complicated. Also, I am not sure what the end result will be.

**Comment by erusev** [comment]
@borekb Which one do you think would work better for you -- `[[toc]]` or a generate command? Thanks.

**Comment by borekb** [comment]
@erusev Probably a command. It can be a complex feature, see e.g. [here](https://github.com/yzhang-gh/vscode-markdown#table-of-contents), so I'd understand if you didn't want to get into it, I just find it useful for longer documents.

**Comment by erusev** [comment]
@borekb Thanks!
